src/test/record/test_frequency.py

```diff
@@ -0,0 +1,169 @@
+            {'timestamp': 3000000000, 'frequency': 1}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=2000000000))
+        assert result == expect_raw
+
+    def test_specify_base_timestamp_case(self):
+        records_raw = [
+            {'timestamp': 5},
+            {'timestamp': 6},
+            {'timestamp': 11},
+            {'timestamp': 12},
+            {'timestamp': 15},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 4, 'frequency': 4},
+            {'timestamp': 14, 'frequency': 1}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10,
+                                              base_timestamp=4))
+        assert result == expect_raw
+
+    def test_two_column_default_case(self):
+        records_raw = [
+            {'timestamp1': 0, 'timestamp2': 2},
+            {'timestamp1': 3, 'timestamp2': 4},
+            {'timestamp1': 11, 'timestamp2': 12},
+            {'timestamp1': 13, 'timestamp2': 14},
+            {'timestamp1': 15, 'timestamp2': 16},
+        ]
+        columns = [ColumnValue('timestamp1'), ColumnValue('timestamp2')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp1': 0, 'frequency': 2},
+            {'timestamp1': 10, 'frequency': 3}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10))
+        assert result == expect_raw
+
+    def test_specify_target_column_case(self):
+        records_raw = [
+            {'timestamp1': 0, 'timestamp2': 2},
+            {'timestamp1': 3, 'timestamp2': 4},
+            {'timestamp1': 11, 'timestamp2': 12},
+            {'timestamp1': 13, 'timestamp2': 14},
+            {'timestamp1': 15, 'timestamp2': 16},
+        ]
+        columns = [ColumnValue('timestamp1'), ColumnValue('timestamp2')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records, target_column='timestamp2')
+
+        expect_raw = [
+            {'timestamp2': 2, 'frequency': 2},
+            {'timestamp2': 12, 'frequency': 3}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10))
+        assert result == expect_raw
+
+    def test_drop_case(self):
+        records_raw = [
+            {'timestamp1': 0, 'timestamp2': 2},
+            {'timestamp2': 4},
+            {'timestamp1': 11, 'timestamp2': 12},
+            {'timestamp1': 13, 'timestamp2': 14},
+            {'timestamp1': 15, 'timestamp2': 16},
+        ]
+        columns = [ColumnValue('timestamp1'), ColumnValue('timestamp2')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp1': 0, 'frequency': 1},
+            {'timestamp1': 10, 'frequency': 3}
```

Comment:
Could you test zero frequency case as well?

```
            {'timestamp1': 0, 'frequency': 1},
            {'timestamp1': 10, 'frequency': 0}
            {'timestamp1': 20, 'frequency': 3}
```

Reply:
I have added a test case and fixed to support this case.
https://github.com/tier4/CARET_analyze/pull/116/commits/d90ebd7fe784546ca68aa366495433d72a274b31